### DISCUSSION TITLE
X-HTTP-Method-Override breaks relationship field on AWS CloudFront hosting.

### DISCUSSION THREAD

**Discussion by BarrySaikSoundry** [payloadcms/payload#7248] [General]
### Payload Version

v3.0.0-beta.56

### Node Version

20

### Next.js Version

15-rc

### Describe the Bug

AWS cloudfront removes the X-HTTP-Method-Override header from HTTP requests as a security measure. In PR#6487 https://github.com/payloadcms/payload/pull/6487 a feature was added for what looks like an edge case of having a large number of items in a relationship field which uses the X-HTTP-Method-Override to tunnel GET requests in POST. This breaks the loading of items and the use of the relationship field on cloudfront hosting, which is also used by AWS amplify. 

### Reproduction Steps

Create a simple relationship between two collections, deploy your project on AWS Amplify. When you try to select the pick list to input the relationship fields you will see the list of items does not load:
<img width="1066" alt="image" src="https://github.com/payloadcms/payload/assets/87904783/1d362257-719c-4248-bca5-5105d466aa46">
<img width="1156" alt="image" src="https://github.com/payloadcms/payload/assets/87904783/784770fc-8c21-477d-8ab1-04bd63a0f1d3">

**Comment by owodunni** [comment]
I have been struggling with this issue for a couple of days. It took me a while to figure out that the `X-Http-Method-Override: GET` header gets removed when calling our kubernetes cluster. We still haven't debugged what part of the cluster is removing it.

But to me it seams a bit unnecessary to use it in:
* [fields/Relationship](https://github.com/payloadcms/payload/blob/beta/packages/ui/src/fields/Relationship/index.tsx#L212)
* [fields/Relationship](https://github.com/payloadcms/payload/blob/beta/packages/ui/src/fields/Relationship/index.tsx#L341)
* [fields/Upload](https://github.com/payloadcms/payload/blob/beta/packages/ui/src/fields/Upload/HasMany/index.tsx#L93)

**Comment by owodunni** [comment]
I made a PR addressing this https://github.com/payloadcms/payload/pull/8017

**Comment by mikexavier** [comment]
@owodunni, did you manage to get this working? I tried forwarding the X-Http-Method-Override header, but the relationship field still isn't functioning as expected. 

I'm wondering if I might have overlooked something, or could there be another reason why relationship fields fail with CloudFront in this setup?

**Reply by owodunni** [reply]
I'm running my own fork that uses GET requests instead of POST..

**Reply by mikexavier** [reply]
Did forwarding the X-Http-Method-Override header not work for you either?

**Comment by rubyjasmin** [comment]
For those looking for a quick and easy workaround I was able to make this work by adding the following to the cloudfront function 

```javascript
if(event.request.headers["x-http-method-override"]) {
  event.request.headers["x-payload-method-override"] = event.request.headers["x-http-method-override"];
  delete event.request.headers["x-http-method-override"];
}
```

Then I added a next.js middleware that converts the x-payload-method-override back to x-http-method-override

```javascript
export function middleware(request: NextRequest) {
  if (!request.headers.has("X-Payload-Method-Override")) {
    return NextResponse.next();
  }
  const headers = new Headers(request.headers);
  headers.set("X-HTTP-Method-Override", "GET");
  return NextResponse.next({
    request: { headers },
  });
}

export const config = {
  matcher: "/api/:path*",
};
```
